metadata-ingestion/src/datahub/sql_parsing/tool_meta_extractor.py

```diff
@@ -73,24 +69,40 @@ def __init__(self, looker_user_mapping: Optional[Dict[str, str]] = None) -> None
     @classmethod
     def create(
         cls,
-        config: ToolMetaExtractorConfig,
         graph: Optional[DataHubGraph] = None,
+        report: ToolMetaExtractorReport = ToolMetaExtractorReport(),
     ) -> "ToolMetaExtractor":
         looker_user_mapping: Dict[str, str] = {}
-        if graph and config.looker_platform_details:
-            key = PlatformResourceKey(
-                platform="looker",
-                resource_type=ResourceType.USER_ID_MAPPING,
-                platform_instance=config.looker_platform_details.platform_instance,
-                primary_key=config.looker_platform_details.platform_instance
-                if config.looker_platform_details.platform_instance
-                else "",
-            )
 
-            platform_resource = PlatformResource.from_datahub(
-                key=key, graph_client=graph
+        if graph:
+            query = (
+                ElasticPlatformResourceQuery.create_from()
+                .group(LogicalOperator.AND)
+                .add_field_match(PlatformResourceSearchFields.PLATFORM, "looker")
+                .add_field_match(
+                    PlatformResourceSearchFields.RESOURCE_TYPE,
+                    ResourceType.USER_ID_MAPPING,
+                )
+                .end()
             )
 
+            platform_resources = [
+                platform_resource
+                for platform_resource in PlatformResource.search_by_filters(
+                    query=query, graph_client=graph
+                )
+            ]
+
+            if platform_resources and sum(1 for _ in platform_resources) > 1:
+                report.warning(
+                    "Tool metadata extraction failed due to multiple platform resources found for Looker."
+                )
+                raise ValueError(
+                    "Exactly one looker platform resource should be present"
+                )
```

Comment:
Why not only check if len(platform_resources)>1?

This should not be fatal error and rest of tool extractor flow should work as expected. Reporting warning is good enough.

"Tool metadata extraction failed due to multiple platform resources found for Looker."  => "Looker user metadata extraction failed. Found more than one looker user id mappings."